src/Navigator.js

```diff
@@ -438,8 +438,9 @@ _flow.transitionIn = function () {
             respondersToWaitFor.push(responder);
 
                 // use namespace transition;
+            const navigator = this instanceof Navigator ? this : null;
             responder.transitionIn(
-                new TransitionCompleteDelegate(responder, TransitionStatus.SHOWN, NavigationBehaviors.SHOW, this, _transition).call
```

Comment:
I could not for the life of me confirm the context of `this` to be an instance of Navigator, and neither could flow. I doubt this was ever working as expected. 

@bassettsj Do you have any ideas? I added a type-check and updated the flow declarations for the `TransitionCompleteDelegate` constructor, but I'm slightly worried this will cause a regression.